Sources/TuistGenerator/Generator/ProjectGroups.swift

```diff
@@ -25,13 +25,14 @@ enum ProjectGroupsError: FatalError, Equatable {
 class ProjectGroups {
     // MARK: - Attributes
 
-    let main: PBXGroup
+    private let main: PBXGroup // main should be accessed from buildMain
     let products: PBXGroup
```

Comment:
`products`, `frameworks` and `playgrounds` should be accessed in a similar way to `main` but I could not a find simple way to do it in this pull request given the use cases we have